### PR TITLE
Use a JUnit 5 `CloseableResource` to reset system properties

### LINKED ISSUE #27996: Test profile config properties can bleed into other test profiles
### Describe the bug

I have run into the problem (I've seen this in dev-mode, but don't know if it is specific to that), that config properties which are provided by test profiles (using `QuarkusTestProfile#getConfigOverrides()`) can bleed into later tests which don't use that profile.

I did some debugging and AFAICT the problem is in `io.quarkus.test.junit.QuarkusTestExtension#doJavaStart()`. In the called `createAugmentor()` method the `RestorableSystemProperties.setProperties()` method is called which sets the profile's config overrides as system properties and also returns a shutdown handler to restore the previous values. But `doJavaStart()` doesn't register these shutdown handlers until the very end and when executing the tests it is possible that these throw an exception (in my case I got an `ExceptionInInitializerError` which as its cause had an `RuntimeException` with the message `Failed to start quarkus`. As a consequence the shutdown handlers never get registered and the profile's config properties remain available as system properties.

### Expected behavior

If profile config property overrides get set as system properties it must be ensured that these get properly reset / cleared, even if the test failed or throws exceptions.

### Actual behavior

_No response_

### How to Reproduce?

If requested I could look into provide a reproducer, but this may take some time.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.12.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by knutwannheden**
Instead of reverting the system properties which may get set by a Quarkus test as part of the `ExtensionState`'s `close()` operation, introduce a new `ShutdownTasks` resource which gets closed independently. Thus, it can perform tasks independently of whether the Quarkus application booted successfully or not.

Fixes #27996

### CODE REVIEW

**Comment by knutwannheden** [comment]
@geoand I found the entire test state management logic to be rather hard to follow and wasn't able to fix this issue in a very simple manner. I also don't know if my PR makes the logic any easier to follow, but this was the best I could come up with now. Please let me know if you can think of a better or simpler idea.

Since not everything will get executed in the same order anymore, we will have to see if there are integration tests that now fail. Also, I would still like to try to implement a test for the issue I was having with the config overrides of test profiles.

**Comment by geoand** [comment]
The test failure do seem related

**Comment by knutwannheden** [comment]
I am on vacation this week, but will look into it again next week.

**Comment by gsmet** [comment]
@knutwannheden I'm closing this PR as it's way outdated. If you want to pursue this, please create another PR. Thanks!
